wiki-service/src/main/java/org/exoplatform/wiki/mow/core/api/wiki/PageImpl.java

```diff
@@ -601,13 +602,27 @@ public synchronized String addRelatedPage(PageImpl page) throws Exception {
       while (refferedIter.hasNext()) {
         Entry<String, Value> entry = refferedIter.next();
         PageImpl page = chSession.findById(PageImpl.class, entry.getValue().getString());
-        if(page != null && page.hasPermission(PermissionType.VIEWPAGE)){
-          relatedPages.add(page);
+        if(!isRelatedPageRemoved(page)){

```

Comment:
- The function getRelatedPages() in PageImpl was called to display a list of related pages on  UIRelatedPagesContainer.gtmpl.
- Function getRelatedPages() get related pages by referenceUUID. But when deleting a page, the referenceUUID wasn't deleted.